src/sample/courseRegController.java

```diff
@@ -229,67 +338,8 @@ public void removecourse() {
       // filter out courses that have been registered by the student.
       updatebothlist();
 
-      // Clear the Course Details after a course is registered.
-      courseTitle.setText("");
-      time.setText("");
-      desc.setText("");
-    }
-  }
-
-  public void searchcourse() {
-    // Get the list of courses that can be registered by the student and stored in
-    // an ArrayList named "list"
-    List<String> list = new ArrayList<>();
-    try {
-      File fileObj = new File("data/Student Course List/" + stuID + ".txt");
-      Scanner fileReader = new Scanner(fileObj);
-      while (fileReader.hasNextLine()) {
-        list.add(fileReader.nextLine());
-      }
-      fileReader.close();
-    } catch (FileNotFoundException e) {
-      System.out.println("An error occurred.");
-      e.printStackTrace();
-    }
-
-    // Perform the search(filter) in "list" and output the result in another
-    // ArrayList named "result"
-    List<String> result = list
-        .stream()
-        .filter(x -> x.toLowerCase().contains(searchCourse.getText().toLowerCase()))
-        .collect(Collectors.toList());
-
-    // Convert the elements in "result" from ArrayList to String
-    StringBuilder strbul = new StringBuilder();
-    for (String str : result) {
-      strbul.append(str);
-      strbul.append("\n");
-    }
-
-    String str = strbul.toString();
-
-    // Clear the Course List ListView
-    courseList.getItems().clear();
-
-    // Insert Course List Search Result into txt file
-    try (FileWriter myWriter = new FileWriter("data/Search.txt", true)) {
-      myWriter.write(str);
-    } catch (IOException e) {
-      System.out.println("An error occurred.");
-      e.printStackTrace();
-    }
-    // Display the search result in Listview
-    try {
-      File fileObj = new File("data/Search.txt");
-      Scanner fileReader = new Scanner(fileObj);
-      while (fileReader.hasNextLine()) {
-        courseList.getItems().add(fileReader.nextLine());
-      }
-      fileReader.close();
-      fileObj.delete();
-    } catch (FileNotFoundException e) {
-      System.out.println("An error occurred.");
-      e.printStackTrace();
+      addCourseButton.setDisable(true);
+      removeCourseButton.setDisable(true);
```

Comment:
why disable the buttons after removing courses? next time someone want to remove then cannot add course ady

Reply:
Here I put `.setDisable(true);` but then if the user selected a course again, then it will become `.setDisable(false);`
You can see `.setDisable(false);` in the `details` function. I tried le it works as expected.